src/include/parser/select_statement.h

```diff
@@ -297,69 +307,53 @@ class SelectStatement : public SQLStatement {
    * @param order_by order by condition
    * @param limit limit condition
    */
-  SelectStatement(std::vector<std::shared_ptr<AbstractExpression>> select, const bool &select_distinct,
-                  std::shared_ptr<TableRef> from, std::shared_ptr<AbstractExpression> where,
-                  std::shared_ptr<GroupByDescription> group_by, std::shared_ptr<OrderByDescription> order_by,
-                  std::shared_ptr<LimitDescription> limit)
+  SelectStatement(std::vector<common::ManagedPointer<AbstractExpression>> select, const bool &select_distinct,
+                  std::unique_ptr<TableRef> from, common::ManagedPointer<AbstractExpression> where,
+                  std::unique_ptr<GroupByDescription> group_by, std::unique_ptr<OrderByDescription> order_by,
+                  std::unique_ptr<LimitDescription> limit)
       : SQLStatement(StatementType::SELECT),
         select_(std::move(select)),
         select_distinct_(select_distinct),
         from_(std::move(from)),
-        where_(std::move(where)),
+        where_(where),
         group_by_(std::move(group_by)),
         order_by_(std::move(order_by)),
         limit_(std::move(limit)),
         union_select_(nullptr) {}
 
   ~SelectStatement() override = default;
 
-  /**
-   * Default constructor for deserialization
-   */
+  /** Default constructor for deserialization. */
   SelectStatement() = default;
 
   void Accept(SqlNodeVisitor *v) override { v->Visit(this); }
 
-  /**
-   * @return select columns
-   */
-  std::vector<std::shared_ptr<AbstractExpression>> GetSelectColumns() { return select_; }
+  /** @return select columns */
+  const std::vector<common::ManagedPointer<AbstractExpression>> &GetSelectColumns() { return select_; }
 
-  /**
-   * @return true if "SELECT DISTINCT", false otherwise
-   */
+  /** @return true if "SELECT DISTINCT", false otherwise */
   bool IsSelectDistinct() { return select_distinct_; }
 
-  /**
-   * @return table being selected from
-   */
-  std::shared_ptr<TableRef> GetSelectTable() { return from_; }
+  /** @return table being selected from */
+  common::ManagedPointer<TableRef> GetSelectTable() { return common::ManagedPointer(from_); }
 
-  /**
-   * @return select condition
-   */
-  std::shared_ptr<AbstractExpression> GetSelectCondition() { return where_; }
+  /** @return select condition */
+  common::ManagedPointer<AbstractExpression> GetSelectCondition() { return common::ManagedPointer(where_); }
```

Comment:
This is already stored as a ManagedPointer, I don't think the wrapper is necessary.

Reply:
I think you think right.